base/src/main/java/com/smartdevicelink/managers/screen/BaseAlertManager.java

```diff
@@ -177,11 +160,30 @@ public void presentAlert(AlertView alert, AlertCompletionListener listener) {
             nextCancelId++;
         }
 
-        PresentAlertOperation operation = new PresentAlertOperation(internalInterface, alert, currentWindowCapability, speechCapabilities, fileManager.get(), nextCancelId, listener);
+        PresentAlertOperation operation = new PresentAlertOperation(internalInterface, alert, currentWindowCapability, speechCapabilities, fileManager.get(), nextCancelId, listener, new AlertSoftButtonClearListener() {
+            @Override
+            public void onButtonClear(List<SoftButtonObject> softButtonObjectList) {
+                // Stop keeping track of SoftButtonObject listeners as operation has finished
+                for (SoftButtonObject object : softButtonObjectList) {
+                    if (softButtonObjects.contains(object)) {
```

Comment:
the if check here is not needed, 

if remove() does not find the object the list is unchaged